tools/noopfactory/src/test/kotlin/com/datadog/tools/noopfactory/NoOpFactoryAnnotationProcessorTest.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.tools.noopfactory
+
+import com.tschuchort.compiletesting.KotlinCompilation
+import com.tschuchort.compiletesting.SourceFile
+import java.io.File
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.CsvSource
+
+internal class NoOpFactoryAnnotationProcessorTest {
+
+    @ParameterizedTest
+    @CsvSource(
+        delimiter = ':',
+        value = [
```

Comment:
maybe we should add also an UnitTest for the case where you have an Interface which inherits from a super one.